clusterloader2/go.mod

```diff
@@ -0,0 +1,72 @@
+module k8s.io/perf-tests/clusterloader2
+
+go 1.12
+
+replace k8s.io/api => k8s.io/api v0.16.4
+
+replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.16.4
+
+replace k8s.io/apimachinery => k8s.io/apimachinery v0.16.4
+
+replace k8s.io/apiserver => k8s.io/apiserver v0.16.4
+
+replace k8s.io/cli-runtime => k8s.io/cli-runtime v0.16.4
+
+replace k8s.io/client-go => k8s.io/client-go v0.16.4
+
+replace k8s.io/cloud-provider => k8s.io/cloud-provider v0.16.4
+
+replace k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.16.4
+
+replace k8s.io/code-generator => k8s.io/code-generator v0.16.4
+
+replace k8s.io/component-base => k8s.io/component-base v0.18.0
+
+replace k8s.io/cri-api => k8s.io/cri-api v0.16.4
+
+replace k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.16.4
+
+replace k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.16.4
+
+replace k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.16.4
+
+replace k8s.io/kube-proxy => k8s.io/kube-proxy v0.16.4
+
+replace k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.16.4
+
+replace k8s.io/kubectl => k8s.io/kubectl v0.16.4
+
+replace k8s.io/kubelet => k8s.io/kubelet v0.16.4
+
+replace k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.16.4
+
+replace k8s.io/metrics => k8s.io/metrics v0.16.4
+
+replace k8s.io/node-api => k8s.io/node-api v0.16.4
+
+replace k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.16.4
+
+replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.16.4
+
+replace k8s.io/sample-controller => k8s.io/sample-controller v0.16.4
+
+replace k8s.io/kubernetes => k8s.io/kubernetes v1.16.4
+
+require (
+	github.com/go-errors/errors v1.0.1
+	github.com/onsi/ginkgo v1.8.0
+	github.com/prometheus/client_model v0.2.0
+	github.com/prometheus/common v0.4.1
+	github.com/spf13/pflag v1.0.5
+	github.com/stretchr/testify v1.4.0
+	golang.org/x/crypto v0.0.0-20190611184440-5c40567a22f8
+	golang.org/x/sync v0.0.0-20190423024810-112230192c58
+	golang.org/x/time v0.0.0-20181108054448-85acf8d2951c
+	gopkg.in/yaml.v2 v2.2.4
+	k8s.io/api v0.16.4
+	k8s.io/apimachinery v1.16.4
```

Comment:
hmm - why here it's 1.16.4 and in replace section 0.16.4?

Why do we even need replace ones? Can't we put everything here?

Reply:
@wojtek-t some of the versioning does the 1.x.y scheme and some do the 0.x.y scheme. Also, starting the initial go mod project is difficult because if you try to "go get" or "go mod download" the bits of the Kubernetes project they try to pull in dependencies from the wrong/mismatched places. There's a lot of discussion in https://github.com/kubernetes/kubernetes/issues/79384#issuecomment-521493597 about this, and there's a script there that many people use to start a project on the desired k8s version.

If I straight try to download the apimachinery 0.16.4 version, for example, we get:
```
% go get k8s.io/apimachinery@v0.16.4
go: finding k8s.io/apimachinery v0.0.0
go: k8s.io/apimachinery@v0.0.0: unknown revision v0.0.0
go: error loading module requirements
%
```

It's possible this is all because we're on 1.12 and these kinds of errors go away once we upgrade to >= Go 1.13

Reply:
Thanks for explaining, @coltonstapper.
FTR, there is nothing prohibiting us to upgrade to go1.13 even now. There is WIP to upgrade k/k repo to go1.14, so we could even target 1.14. That being said, I'm totally fine with starting with 1.12 and upgrading later. Thanks for doing this! This is great!

/lgtm
/hold 
Leaving for Wojtek for a final approval and lift of the hold

Reply:
Thanks @mm4tt I'll be fast-following with the Golang upgrade so we can be getting to a newer version of Kubernetes as well.

Reply:
I hate the fact how many dependencies we have (e.g. why we're relying on sample-apiserver - that sounds crazy. Event the dependency on apiserver itself is crazy - we should just depend on client-go and apimachinery and the API itself).
Anyway - that's for another discussion...